crates/nu-std/lib/mod.nu

```diff
@@ -108,13 +108,17 @@ def check-clipboard [
 export def clip [
     --silent: bool  # do not print the content of the clipboard to the standard output
     --no-notify: bool  # do not throw a notification (only on linux)
+    --no-strip: bool  # do not strip ANSI escape sequences from a string
     --expand (-e): bool  # auto-expand the data given as input
 ] {
     let input = (
         $in
         | if $expand { table --expand } else { table }
         | into string
-        | ansi strip
+    )
+    
+    let input = (
+        if $no_strip { $input } else { $input | ansi strip }
```

Comment:
Seems like this may be better replacing `ansi strip` above instead of shadowing `input`. Is that not possible?

Reply:
@maxim-uvarov 
you should be able to just write
```
    let input = (
        $in
        | if $expand { table --expand } else { table }
        | into string
        | if $no_strip {} else { ansi strip }
```
:wink:

Reply:
Agreed. That's what I meant. :) Thanks!

Reply:
Wow! It works! @antoine, @fdncred, thank you! 

However, I don't understand how it works. Could you please explain it to me or direct me to where I can read about it? 

I mean, I don't understand how the 'if' command behaves in the pipe.

Reply:
wow, now I see that this pattern is used in many places in std. Cool! 🙏

Reply:
i think `| if ...` consumes the input stream of data and passes it to one of its branches.
when you write `$d | if $cond { a } else { b }`, if `$cond` evaluates to `true`, `a` is applied with `$d` as the input, otherwise, it's `b` :ok_hand: 

finally there is the special case where one of the branch is empty, i.e. `{}`, which is equivalent to `{ $in }` and thus simply passes `$d` down the pipeline.

i'm no expert on this, i hope it's both helpful and kinda correct :pray:

Reply:
@amtoine, thank you so much! 🙏

I believe it's worth adding this information to the book as it is a very frequent and useful feature. I can make a PR by myself in a few days, or if you or somebody else feel up to it, that would be cool too!

Reply:
please go ahead that's a great idea :blush: 
that might a good thing to see a member of the community explaining it, making sure it's clear how this works :smirk: